source/material_model/depth_dependent.cc

```diff
@@ -0,0 +1,444 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/material_model/depth_dependent.h>
+#include <deal.II/base/parameter_handler.h>
+#include <utility>
+#include <limits>
+using namespace dealii;
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+
+    namespace internal
+    {
+      class RadialViscosityLookup
+      {
+        public:
+          RadialViscosityLookup(const std::string &filename)
+          {
+            /* This method is used for the Table method of depth dependent viscosity */
+            std::string temp;
+            std::ifstream in(filename.c_str(), std::ios::in);
+            AssertThrow (in,
+                         ExcMessage (std::string("Couldn't open file <") + filename));
+
+            min_depth=DBL_MAX;
+            max_depth=DBL_MIN;
+
+            // The input viscosity file has two columns that are the viscosity and Depth
+            std::string header;
+            getline(in, header);/* Discard header line */
+            while (!in.eof())
+              {
+                double visc, depth;
+                in >> visc;;
+                if (in.eof())
+                  break;
+                in >> depth;
+                getline(in, temp);
+
+                min_depth = std::min(depth, min_depth);
+                max_depth = std::max(depth, max_depth);
+
+                values.push_back(visc);
+                depthvalues.push_back(depth);
+              }
+          }
+
+          double radial_viscosity(double depth)
+          {
+            // Interpolate the viscosity to the nearest neighbour
+            std::vector<double> depth_diff (values.size(), 0);
```

Comment:
This must surely be the most expensive way to implement this algorithm :-)

I think what you really want is to look at the `InterpolatedTensorProductGridData` in deal.II (with dim==1) that provides this sort of functionality already: all you have to do is give it a list of depth points and values at these points, and it will automatically interpolate between them. In fact, you can even choose to interpolate linearly, rather than taking the value from the closest data point as you do here.